packages/superset-ui-chart/src/models/ChartProps.ts

```diff
@@ -1,30 +1,28 @@
 import { createSelector } from 'reselect';
 import { convertKeysToCamelCase } from '@superset-ui/core';
+import { Datasource } from '@superset-ui/query';
 import { HandlerFunction, PlainObject } from '../types/Base';
+import { QueryData, DataRecordFilters } from '../types/QueryResponse';
 
 // TODO: more specific typing for these fields of ChartProps
 type AnnotationData = PlainObject;
-type CamelCaseDatasource = PlainObject;
 type SnakeCaseDatasource = PlainObject;
 type CamelCaseFormData = PlainObject;
 type SnakeCaseFormData = PlainObject;
-export type QueryData = PlainObject;
-/** Initial values for the visualizations, currently used by only filter_box and table */
-type InitialValues = PlainObject;
+type RawFormData = CamelCaseFormData | SnakeCaseFormData;
+
 type ChartPropsSelector = (c: ChartPropsConfig) => ChartProps;
 
 /** Optional field for event handlers, renderers */
 type Hooks = {
-  /** handle adding filters  */
-  onAddFilter?: HandlerFunction;
+  onAddFilter?: (newFilters: DataRecordFilters, merge?: boolean) => void;
```

Comment:
should we add back the doc string?

Reply:
The doc string doesn't seem very helpful to me. `onAddFilter` is basically synonymous to `handle adding filter`.